core/src/types.rs

```diff
@@ -399,13 +436,15 @@ impl Default for SyncClientConfig {
 pub struct AppClientConfig {
 	pub disable_rpc: bool,
 	pub threshold: usize,
+	pub network_mode: NetworkMode,
 }
 
 impl Default for AppClientConfig {
 	fn default() -> Self {
 		Self {
 			disable_rpc: false,
 			threshold: 5000,
+			network_mode: NetworkMode::Both,
```

Comment:
We can remove `disable_rpc` entirely and use equality directly in the code, since we are storing network_mode in the configuration.